### PR TITLE
feat: Adds message signature for new email conversations

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

## Description

This PR allows setting message signatures for newly created email conversations from the new conversation modal.

Fixes https://linear.app/chatwoot/issue/CW-2376/add-signature-for-new-email-conversations

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

**Loom video**
https://www.loom.com/share/fd9f2ca89109453a96826331ac1dba00?sid=0e2a147e-ae9e-4b92-b092-2073330270fa

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/dashboard/routes/dashboard/conversation/contact/ConversationForm.vue
Comment (iamsivin) [inline app/javascript/dashboard/routes/dashboard/conversation/contact/ConversationForm.vue:107]: Moved out the label because it caused some weird click issues when we clicked the editor placeholder text.

**Screen recordings**

https://github.com/chatwoot/chatwoot/assets/64252451/a960b699-1885-403e-989d-df4e82f9bf2e

**Comment by iamsivin** [comment]
Only merge this PR after merging this [PR](https://github.com/chatwoot/chatwoot/pull/7944)

File: app/javascript/dashboard/components/widgets/conversation/MessageSignature.vue
Comment (scmmishra) [inline app/javascript/dashboard/components/widgets/conversation/MessageSignature.vue:5]: What does the `!m-0` do?

File: app/javascript/dashboard/components/widgets/WootWriter/Editor.vue
Comment (scmmishra) [inline app/javascript/dashboard/components/widgets/WootWriter/Editor.vue:26]: Can we name this slot to `footer` or something

File: app/javascript/dashboard/routes/dashboard/conversation/contact/ConversationForm.vue
Comment (scmmishra) [inline app/javascript/dashboard/routes/dashboard/conversation/contact/ConversationForm.vue:255]: No need to clean signature here, it's already done when applying or removing

File: app/javascript/dashboard/routes/dashboard/conversation/contact/ConversationForm.vue
Comment (scmmishra) [inline app/javascript/dashboard/routes/dashboard/conversation/contact/ConversationForm.vue:297]: Are we enabling it only for email?

**Review by scmmishra — CHANGES_REQUESTED** [review]

File: app/javascript/dashboard/routes/dashboard/conversation/contact/ConversationForm.vue
Comment (iamsivin) [reply app/javascript/dashboard/routes/dashboard/conversation/contact/ConversationForm.vue:297]: Yeah, I think so. The linear issue is based on adding a signature for a new email conversation. If we are planning, I can create a separate PR.

File: app/javascript/dashboard/components/widgets/conversation/MessageSignature.vue
Comment (iamsivin) [reply app/javascript/dashboard/components/widgets/conversation/MessageSignature.vue:5]: @scmmishra I added it because the `p` tag has some base styles that will not be removed when we add like this `m-0`. That's the reason I added it as important.

File: app/javascript/dashboard/components/widgets/conversation/MessageSignature.vue
Comment (iamsivin) [reply app/javascript/dashboard/components/widgets/conversation/MessageSignature.vue:1]: Moved the file name to `MessageSignatureMissingAlert` https://github.com/chatwoot/chatwoot/pull/7946/commits/47bb845e58dec10997bde626b018e0ba060b8574

File: app/javascript/dashboard/routes/dashboard/conversation/contact/ConversationForm.vue
Comment (iamsivin) [reply app/javascript/dashboard/routes/dashboard/conversation/contact/ConversationForm.vue]: Done https://github.com/chatwoot/chatwoot/pull/7946/commits/47bb845e58dec10997bde626b018e0ba060b8574

File: app/javascript/dashboard/routes/dashboard/conversation/contact/ConversationForm.vue
Comment (iamsivin) [reply app/javascript/dashboard/routes/dashboard/conversation/contact/ConversationForm.vue]: Closing this as it's fixed in a separate PR

File: app/javascript/dashboard/components/widgets/conversation/MessageSignature.vue
Comment (scmmishra) [reply app/javascript/dashboard/components/widgets/conversation/MessageSignature.vue]: Cool, what is the source of the style?

Comment (iamsivin) [reply app/javascript/dashboard/components/widgets/conversation/MessageSignature.vue]: @scmmishra Removing the default margin bottom from the paragraph.

**Review by scmmishra — APPROVED** [review]

**Pull request merged**
